src/Components/Login/Login.tsx

```diff
@@ -1,76 +1,111 @@
           <label htmlFor="email" className="w-full ps-1 mb-1">
-          Registered email address
+            Registered email address
           </label>
-          <div className="flex rounded-md border-3 border-white">
+          <div             className={`flex rounded-md border-3 ${
+              !errors.email ? "border-white" : "border-red-500"
+            }`}>
             <span className="flex select-none items-center me-3 pl-3 text-white ">
               <i className="fa-solid fa-envelope"></i>
             </span>
             <input
+              {...register("email", {
+                required: "Email is required",
+                pattern: {
+                  value: /^[A-Z0-9.]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
+                  message: "Invalid email address",
+                },
+              })}
               type="text"
               id="email"
               className="block px-2  flex-1 border-0 bg-transparent py-1.5 pl-1 text-white placeholder:text-gray-400 focus:ring-0 sm:text-sm sm:leading-6"
               placeholder="Type your email"
             />
+                        {errors.email && (
+              <AlertMessage message={String(errors.email.message)}/>
+            )}
           </div>
         </div>
 
         <div className="password mt-2">
           <label htmlFor="password" className="w-full ps-1 mb-1">
             Password
           </label>
-          <div className="flex rounded-md border-3 border-white">
+          <div className={`flex rounded-md border-3 ${
+              !errors.password ? "border-white" : "border-red-500"
+            }`}>
             <span className="flex select-none items-center me-3 pl-3 text-white ">
               <i className="fa-solid fa-key"></i>
             </span>
             <input
-              type="text"
+             {...register('password', {
+              required: 'field is required',
+              minLength: {
+                value: 8,
+                message: 'Password should be at least 8 characters',
+              },
+            })}
+              type="password"
               id="password"
               className="block px-2  flex-1 border-0 bg-transparent py-1.5 pl-1 text-white placeholder:text-gray-400 focus:ring-0 sm:text-sm sm:leading-6"
               placeholder="Type your password"
             />
+             {errors.password && (
+                          <AlertMessage message={String(errors.password.message)}/>
+            )}
           </div>
         </div>
 
         <div className="flex mt-3 mb-16 justify-between items-end">
-        <button
-          type="submit"
-          className="bg-slate-50 transition duration-100 hover:bg-gray-800  text-slate-950  hover:text-slate-50  rounded-lg px-4 py-2 font-medium "
-        >
-          Sign In{" "}
-          <i className="fa-solid fa-check ms-1 text-white p-1 rounded-full bg-black"></i>
-        </button>
+        <AuthButton text={String("Sign In")}/>
```

Comment:
I don't understand the reason for using `String` here and in other places of the code as well. Could you explain the reason for that?